src/adapters/ActivitiesJSONAdapter.js

```diff
@@ -59,4 +69,40 @@ export default class ActivitiesJSONAdapter extends ActivitiesAdapter {
       observer.complete();
     });
   }
+
+  /**
+   * Posts an attachment action, returns an observable that emits the created action
+   *
+   * @param {string} activityID  ID of the activity corresponding to this submit action
+   * @param {object} inputs  The message content
+   * @returns {Observable.<object>} Observable stream that emits data of the newly created action
+   */
+  postAction(activityID, inputs) {
+    return Observable.create((observer) => {
+      const activity = this.datasource[activityID];
+
+      if (activity) {
+        const actionID = `action-${activityID}-${this.datasource[activityID]?.actions.length + 1}`;
+        const personID = 'user1';
+        const {roomID} = this.datasource[activityID];
+        const newAction = {
+          ...EMPTY_ACTION,
+          actionID,
+          personID,
+          roomID,
+          activityID,
+          inputs,
+          created: new Date().toISOString(),
+        };
+        const activityActions = this.datasource[activityID].actions;
+
+        observer.next(newAction);
+        activityActions.push(newAction);
+      } else {
+        observer.error(new Error(`Unable to create an attachment action for activity with id "${activityID}"`));
+      }
+
+      observer.complete();
```

Comment:
observer.error() already completes the observable. Better move this inside the 'if'